### ISSUE TITLE
Tell me what is going on with missions

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K#1806]
One of the most vital and (apparently) least complete parts of ZK for new players is the missions. Mainly the chobby interface for missions as well as the existence of good tutorial and introductory missions. What is going on which missions? How are they made? Why are there little-to-no complete ones that I know of?

**Comment by KingRaptor** [comment]
> How are they made?

Currently the only way that infra supports is with the [Zero-K Mission Editor](http://zero-k.info/mediawiki/index.php?title=Mission_Editor). I think only me, Sortale and Anarchid have any real experience with the thing, but it should be reasonably easy for an interested person to learn how to use it if they have an example to look at and follow the instructions on the wiki.

> Why are there little-to-no complete ones that I know of?

Because making missions is Work. (Compare with wiki, which also has seen little work in either mediawiki or avatarwiki forms despite having a much lower time investment required compared to missions)

> What is going on which missions?

I've been working (slowly) on the previously discussed tutorial missions and some things for save/load support. Dunno if anyone else is doing anything.

**Comment by CrazyEddieTK** [comment]
I think, conceptually, we have the following types of missions:

- Tutorials
- Practice - Battle
- Practice - Survival
- Campaign
- Challenge

There are three tutorials (UI, Econ, Combat) and they are all very good. I think they could use some minor improvements. There are five Battle Practice (Sandbox, 2v1, 2v2, 1v1, 1v2) and three Survival Practice (Very Easy, Easy, and Normal). They need some fixes and some added user-friendliness such as intro screens, objectives, and tips.

The tutorials and practice missions together make a great introduction to Zero-K. I would take these eleven missions, give them a bit of polish, and declare things done.

The campaign would be great to have but is not essential. If KingRaptor and Sortale think that their current Work In Progress is good enough, they could be published in chobby - perhaps with a notice that the campaign is still a work in progress.

The rest of the missions are a huge mixed bag, most of them broken or bad. A few of them might well be worth the effort to fix/polish and publish.

In chobby, I would reorganize the Singleplayer tabs. Eliminate Missions and Skirmish, and replace them with:

- Training
- Campaign
- Challenge
- Custom Battle

Under Training, split the screen into three sections: Tutorial, Practice - Battle, and Practice - Survival. In each section show the appropriate missions using large friendly letters, large pictures, and large buttons that make it obvious where to click to play them. Disregard the current UI on the current Missions page; the Training page doesn't need it. Leave out difficulty and map and mission number. Include name, description, picture, and start button.

The Campaign and Challenge screens could each have their own organization scheme as would be most appropriate for them. For instance: difficulty might be appropriate to show for the Campaign missions, or not; it definitely would be appropriate for the Challenge missions.

**_In short:_**

- Split up the missions by type
- The Training missions are ready now
- The Campaign missions could be added now, or later when ready
- The Challenge missions could have a few now, or soon, and more later when people make more

**Comment by GoogleFrog** [comment]
@CrazyEddieTK you should run singleplayer in dev mode to see the current progress of things. I'm doing a campaign instead of KingRaptor and Sortale.

I think Practice (battle and survival) can both be removed because the simple skirmish setup can replace them. All we need in terms of missions is the three tutorials, they can be under a tutorial menu. The third tutorial needs renaming and isn't quite as good as the others in terms of polish. My plan for submenus is:
 * Tutorial
 * Campaign
 * Skirmish
 * Load

**Comment by CrazyEddieTK** [comment]
I think there's value in having the Sandbox -> 2v2 -> 2v1 -> 1v1 sequence presented to new users as an Official Thing You Can (And Probably Should) Do in the user interface, rather than leaving it to their own devices to discover via the skirmish interface.

That might not be true once the campaign is done, since presumably the campaign will serve as a graduated introduction to playing ZK. But until then, encouraging people to learn to play the game via a series of increasingly difficult sandbox environments (2v2 and 2v1 are effectively sandboxes, as your AI ally will keep you alive while you figure out what you're doing) is a very, very good thing to do. I think it may be the most effective way we have right now to teach people the game while keeping them engaged rather than making them discouraged.

Even after the campaign is done, I think there would still be value in having that sequence. People may be interested in learning the game but not interested playing through a campaign.

**Comment by CrazyEddieTK** [comment]
(sidebar: is "Skirmish" a good term? Will new players understand what it means _before_ they click on it and play with the UI to see what it does?)

**Comment by sprunk** [comment]
Anything to do gameside? Looks like this belongs to Chobby.

**Comment by GoogleFrog** [comment]
I think I know what is going on with missions. There are three tutorials, that is all. If more tutorials are produced then they can be fit in.
